src/OBD2/OBD2.c

```diff
@@ -110,12 +116,140 @@ static int decode_pid(unsigned char pid, CAN_msg *msg, int *value)
         case 0x11: //throttle position %
             *value = A * 100 / 255;
             break;
+        case 0x14: //Oxygen sensor voltage - Bank 1, Sensor 1
+        case 0x15: //Oxygen sensor voltage - Bank 1, Sensor 2
+        case 0x16: //Oxygen sensor voltage - Bank 1, Sensor 3
+        case 0x17: //Oxygen sensor voltage - Bank 1, Sensor 4
+        case 0x18: //Oxygen sensor voltage - Bank 2, Sensor 1
+        case 0x19: //Oxygen sensor voltage - Bank 2, Sensor 2
+        case 0x1A: //Oxygen sensor voltage - Bank 2, Sensor 3
+        case 0x1B: //Oxygen sensor voltage - Bank 2, Sensor 4
+            *value = A / 200;
+            break;
+        case 0x1F: //Run time since engine start
+            *value = (A * 256) + B;
+            break;
+        case 0x22: //Fuel rail Pressure (relative to manifold vacuum) kPa
+            *value = ((A * 256) + B) * 0.079;
+            break;
+        case 0x23: //Fuel rail Pressure (diesel, or gasoline direct inject) kPa 
+            *value = ((A * 256) + B) * 10;
```

Comment:
It would be clearer to write `(A << 8 | B)` instead of the math since that is the intention here.